runtime/libs/misc/CMakeLists.txt

```diff
@@ -8,4 +10,23 @@ target_link_libraries(nnfw_lib_misc PRIVATE nnfw_common)
 target_link_libraries(nnfw_lib_misc PRIVATE nnfw_coverage)
 
 add_executable(nnfw_tensor_index_iterator "examples/tensor_index_iterator.cpp")
-target_link_libraries(nnfw_tensor_index_iterator nnfw_lib_misc)
+target_link_libraries(nnfw_tensor_index_iterator PRIVATE nnfw_lib_misc)
+
+# Gtest misc_index_enumerator_test
+if(NOT ENABLE_TEST)
+  return()
+endif(NOT ENABLE_TEST)
+
+add_executable(misc_index_enumerator_test src/IndexEnumerator.test.cpp)
+target_link_libraries(misc_index_enumerator_test PRIVATE nnfw_lib_misc)
+
+nnfw_find_package(GTest)
+if(NOT GTest_FOUND)
+    message(STATUS "GTest not avaialble. Skipping lib/misc test build")
+    return()
+endif(NOT GTest_FOUND)
```

Comment:
If `GTEST_FOUND` is false, `ENABLE_TEST` become false automatically.
So we don't need this check.

https://github.com/Samsung/ONE/blob/master/infra/nnfw/CMakeLists.txt#L85-L91